File: src/main/java/br/unifor/costify/application/dto/entity/RecipeDto.java
package br.unifor.costify.application.dto.entity;

import br.unifor.costify.application.dto.response.RecipeIngredientDto;
import br.unifor.costify.domain.entity.Recipe;
import java.math.BigDecimal;
import java.util.List;

public record RecipeDto(String id, String name, List<RecipeIngredientDto> ingredients, BigDecimal totalCost) {

  public static RecipeDto from(Recipe recipe) {
    List<RecipeIngredientDto> ingredientDtos = recipe.getIngredients().stream()
            .map(RecipeIngredientDto::from)
            .toList();

    return new RecipeDto(
            recipe.getId().getValue(),
            recipe.getName(),
            ingredientDtos,
            recipe.getTotalCost().getAmount()
    );
  }
}

File: src/application/dto/entity/RecipeDto.ts
import { Unit } from '@domain/valueobjects/Unit.js'

export interface RecipeIngredientDto {
  readonly ingredientId: string
  readonly quantity: string
  readonly unit: Unit
}

export interface RecipeDto {
  readonly id: string
  readonly name: string
  readonly ingredients: RecipeIngredientDto[]
  readonly totalCost: string
}
